connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -208,6 +209,21 @@ public static String COMMIT_TASKS_KEY(String connectorName) {
             .field("creation-timestamp", Schema.INT64_SCHEMA)
             .build();
 
+    public static final String RESTART_PREFIX = "restart-connector-";
+
+    public static String RESTART_KEY(String connectorName) {
+        return RESTART_PREFIX + connectorName;
+    }
+
+    public static final Boolean ONLY_FAILED_DEFAULT = Boolean.FALSE;
+    public static final Boolean INCLUDE_TASKS_DEFAULT = Boolean.FALSE;
```

Comment:
any reason not to use primitive types?
```suggestion
    public static final boolean ONLY_FAILED_DEFAULT = false;
    public static final boolean INCLUDE_TASKS_DEFAULT = false;
```

Reply:
The defaults are used in code like below and using primitive would lead to an extra boxing

>         Object failed = valueAsMap.get(ONLY_FAILED_FIELD_NAME);
>         if (failed == null) {
>             log.warn("Invalid data for restart request '{}' field was missing, defaulting to {}", ONLY_FAILED_FIELD_NAME, ONLY_FAILED_DEFAULT);
>             failed = ONLY_FAILED_DEFAULT;
>         }
>         if (!(failed instanceof Boolean)) {
>             log.warn("Invalid data for restart request '{}' field should be a Boolean but is {}, defaulting to {}", ONLY_FAILED_FIELD_NAME, failed.getClass(), ONLY_FAILED_DEFAULT);
>             failed = ONLY_FAILED_DEFAULT;
>         }
>         boolean onlyFailed = (Boolean) failed;

Reply:
I agree the conversion is on a frequently used path. But maybe it's the code below that can be re-written to avoid both autoboxing and unboxing (when it's not required)

Reply:
fixed as per your suggestions.